### ISSUE TITLE
No translations for the contained tables

### ISSUE THREAD

**Issue by DanielPenkov** [AD7six/cakephp-shadow-translate#25]
It seems that translations dont work on the contained tables. If I have `$Forms->get($id)` everything is Ok, but if I have `$Companies->get($id, ['contain' => ['Forms']])` , the table Forms is not translated. If I change the behavior from ShadowTranslate to Translate Im getting the translations. I looked into the function _rowMapper() which should merge the translated fields into each entity for a given locale, but it was not executed for the contained results.

**Comment by Spriz** [comment]
:+1:

**Comment by AD7six** [comment]
$Companies->get($id, ['contain' => ['Forms', 'FormsTranslations']])

Probably just works.

**Comment by DanielPenkov** [comment]
Hi, thank you for your answer!
I tried `$Companies->get($id, ['contain' => ['Forms', 'Forms.FormsTranslations']]);` , but I am still missing the translations.
If I debug the result from the query I can see that the translations are there but they are not mapped to the fields.
```
[
	'id' => 'd8dfeda7',
	'language_id' => 'f29f1d42-b1d3',
	'name' => 'Company name',
	'forms' => [
		(int) 0 => [
			'id' => 'f7a4220b',
			'name' => '',
			'description' => '',
			'_i18n' => [
				(int) 0 => [
					'id' => 'f7a4220b',
					'locale' => 'de_DE',
					'name' => 'Vereinbarungen',
					'description' => null
				]
			],
			'_joinData' => [
				'form_id' => 'f7a4220b',
				'id' => '02f30833',
				'company_id' => 'd8dfeda7',
			]
		]
	]
]
```

In the Forms model I have: 
`$this->addBehavior('ShadowTranslate.ShadowTranslate', ['fields' => ['name', 'description']]);`

My locale is set to 'de_DE'.
I don`t know if I am doing something wrong or there is some bug :)

**Comment by Spriz** [comment]
Any news on this? :smile:

**Comment by MrSnoop** [comment]
I have the same issue.
The behavior works fine. But on the first level of association loaded via contain does not return mapped data. Data is there in _i18n but not mapped like DanielPenkov pointed out. 
And if a deeper associaten is loaded like 

> 'contain'=>'Media.Mediaitems'

it works fine for Mediaitems, Media gets default language output.

**Comment by DanielPenkov** [comment]
@MrSnoop I had a problem with retrieving the fields that need to be translated from the query in addFieldsToQuery method, so I made it always to take all translation fields for the model. You can take a look here: https://github.com/AD7six/cakephp-shadow-translate/pull/32

**Comment by ADmad** [comment]
Fixed in #37
